wgpu-hal/src/dynamic/device.rs

```diff
@@ -0,0 +1,13 @@
+use crate::{Device, DynBuffer};
+
+use super::DynResourceExt;
+
+pub trait DynDevice {
+    unsafe fn destroy_buffer(&self, buffer: Box<dyn DynBuffer>);
+}
+
+impl<D: Device> DynDevice for D {
+    unsafe fn destroy_buffer(&self, buffer: Box<dyn DynBuffer>) {
+        unsafe { D::destroy_buffer(self, buffer.unbox()) };
+    }
+}
```

Comment:
issue: We don't document a safety contract or adherence thereto for these methods. We should do this.